### ISSUE TITLE
num_group not handled in CUDNN backend

### ISSUE THREAD

**Issue by leehonglak** [dmlc/mxnet#341]
Hello,

Thanks a lot for mxnet, it's amazing!

I'm attempting to use `num_group` with the CUDNN backend but I'm having some issues, and I don't see any handling of this parameter in the implementation:
https://github.com/dmlc/mxnet/blob/master/src/operator/cudnn_convolution-inl.h#L170

Am I missing something? Should I only use the Toeplitz/CUBLAS backend?

**Comment by antinucleon** [comment]
In my knowledge, CuDNN doesn't support num_group.  Even in AlexNet2, Alex removed group so feel free to remove group in your network

**Comment by leehonglak** [comment]
I'm actually using `num_group` as a feature ;-) I need it in my networks!

**Comment by leehonglak** [comment]
I was hoping to find something like:
https://github.com/soumith/cudnn.torch/blob/master/SpatialConvolution.lua#L116

I guess an update to the documentation would be welcomed, since `num_group` doesn't seem to work, even with the CUBLAS backend...

**Comment by tqchen** [comment]
hmm, num_group should work in the original version of CONV as far as I know, but we have to admit that this is a feature that we did not use frequently so maybe we miss something here.

@antinucleon Let us do a round of test with num_group, and maybe configure the operator to return non-cudnn based op when it is bigger than 1

**Comment by tqchen** [comment]
BTW @leehonglak Seems what you want is split channels into groups and do convolution separately. I think you can also achieve it by the following code in current symbolic API, which enables us to use CUDNN as well

```python
splited = mx.sym.SplitChannel(in, num_outputs=ngroup)
outs = [mx.sym.Convolution(splited.outputs[i], other parameters) for i in range(ngroup)]
final = mx.sym.Concat(*outs)
```

**Comment by tqchen** [comment]
c.f. http://mxnet.readthedocs.org/en/latest/python/symbol.html#mxnet.symbol.SliceChannel

**Comment by leehonglak** [comment]
@tqchen, your solution is even better (i.e. more flexible) for our purpose, thanks a ton!

**Comment by leehonglak** [comment]
Unfortunately, it looks like SliceChannel's `num_outputs` is limited to max 5 slices. Is that correct? would it be possible to lift this hard constraint?

**Comment by antinucleon** [comment]
It is possible, but because I am too lazy to write more. Also you may consider to slice several times, aka
```
splited = mx.sym.SliceChannel(in, num_outputs=5)
outs = [mx.sym.SliceChannel(splited.outputs[i], num_outputs=5) for i in range(ngroup)]
...
```
to get 25 sliced output

**Comment by tqchen** [comment]
@antinucleon I think we can wrap this behavior, to support arbitrary slice by multiple kernel calls.

**Comment by tqchen** [comment]
@leehonglak The constraint will be lifted to any max slice after #441 get merged

**Comment by tqchen** [comment]
This is resolved as #441 is merged in
